sdk/component.js

```diff
@@ -176,12 +176,16 @@ class Component {
    * @protected
    */
   _replyWith(metadata, payload) {
-    this.busy = true;
-    // To check out the output payload, it's better to use katana service/middleware start --transport
-    // You might need to inspect this if the payload is malformed, though
-    logger.debug('Sending command reply', metadata, JSON.stringify(payload));
-    this.socket.send([metadata, msgpack.pack(payload)]);
-    this.busy = false;
+    if (this.socket) {
```

Comment:
It would be clearer by using a guard clause

```
if (this.socket) {
  return logger.debug('No connection to socket');
}
```

And then the rest

Reply:
OK, I've updated that, just with `!this.socket`.